xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -252,6 +260,180 @@ public FeeResult fee() throws JsonRpcClientErrorException {
     return jsonRpcClient.send(request, FeeResult.class);
   }
 
+  /**
+   * Get the ledger index of a tx result response. If not present, throw an exception.
+   *
+   * @return A string containing value of last validated ledger index.
+   */
+  protected UnsignedInteger getMostRecentlyValidatedLedgerIndex() throws JsonRpcClientErrorException {
+    JsonRpcRequest request = JsonRpcRequest.builder()
+      .method(XrplMethods.LEDGER)
+      .addParams(LedgerRequestParams.builder().ledgerSpecifier(LedgerSpecifier.VALIDATED).build())
+      .build();
+    return jsonRpcClient.send(request, LedgerResult.class).ledgerIndexSafe().unsignedIntegerValue();
+  }
+
+  /**
+   * Get the {@link TransactionResult} for the transaction with the hash transactionHash.
+   * @param transactionHash {@link Hash256} of the transaction to get the TransactionResult for.
+   * @return the {@link TransactionResult} for a validated transaction and empty response for a
+   *        {@link Transaction} that is expired or not found.
+   */
+  protected Optional<? extends TransactionResult<? extends Transaction>> getValidatedTransaction(
+    final Hash256 transactionHash
+  ) {
+    Objects.requireNonNull(transactionHash);
+
+    try {
+      final TransactionResult<? extends Transaction> transactionResult = this.transaction(
+        TransactionRequestParams.of(transactionHash), Transaction.class
+      );
+      return Optional.ofNullable(transactionResult).filter(TransactionResult::validated);
+    } catch (JsonRpcClientErrorException e) {
+      // The transaction was not found on ledger, warn on this, but otherwise return.
+      LOGGER.warn(e.getMessage(), e);
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Check if there missing ledgers in rippled in the given range.
+   * @param submittedLedgerSequence {@link LedgerIndex} at which the {@link Transaction} was submitted on.
+   * @param lastLedgerSequence he ledger index/sequence of type {@link UnsignedInteger} after which the
+   *        transaction will expire and won't be applied to the ledger.
+   * @return {@link Boolean} to indicate if there are gaps in the ledger range.
+   */
+  protected boolean ledgerGapsExistBetween(UnsignedLong submittedLedgerSequence, UnsignedLong lastLedgerSequence) {
```

Comment:
Let's add unit tests for this method directly (i.e., `ledgerGapsExistBetween`). We should test a variety of inputs here (even nulls) against a variety of `serverInfo` responses (likely means you'll need to mock `jsonRpcClient`).